src/hooks/useProductQuery.tsx

```diff
@@ -1,12 +1,10 @@
-import { useQuery } from 'react-query'
-import ky from 'ky'
+import withQuery from './withQuery'
 
-function useProductQuery(productId: string) {
-  return useQuery('product', async () => {
-    const response = ky.get(`https://fakestoreapi.com/products/${productId}`)
-    const data: ProductItem = await response.json()
-    return data
-  })
+function useProductQuery(productId: string | undefined) {
+  return withQuery<ProductItem>(
+    `product-${productId}`,
```

Comment:
I think it's just a better standard to have array query keys. Allows you to potentially do things like invalidate all caches of product queries  
```suggestion
    ['product', productId],
```